recOrder/io/visualization.py

```diff
@@ -0,0 +1,164 @@
+import numpy as np
+
+from typing import Literal, Union
+from colorspacious import cspace_convert
+from matplotlib.colors import hsv_to_rgb
+from skimage.color import hsv2rgb
+from skimage.exposure import rescale_intensity
+
+
+def ret_ori_overlay(
+    czyx,
+    ret_max: Union[float, Literal["auto"]] = 10,
+    cmap: Literal["JCh", "HSV"] = "JCh",
+):
+    """
+    Creates an overlay of retardance and orientation with two different colormap options.
+    "HSV" maps orientation to hue and retardance to value with maximum saturation.
+    "JCh" is a similar colormap but is perceptually uniform.
+
+    Parameters
+    ----------
+    czyx:                   (nd-array) czyx[0] is retardance in nanometers, czyx[1] is orientation in radians [0, pi],
+                            czyx.shape = (2, ...)
+
+    ret_max:                (float) maximum displayed retardance. Typically use adjusted contrast limits.
+
+    cmap:                   (str) 'JCh' or 'HSV'
+
+    Returns
+    -------
+    overlay                 (nd-array) RGB image with shape (3, ...)
+
+    """
+    if czyx.shape[0] != 2:
+        raise ValueError(
+            f"Input must have shape (2, ...) instead of ({czyx.shape[0]}, ...)"
+        )
+
+    retardance = czyx[0]
+    orientation = czyx[1]
+
+    if ret_max == "auto":
+        ret_max = np.percentile(np.ravel(retardance), 99.99)
+
+    # Prepare input and output arrays
+    ret_ = np.clip(retardance, 0, ret_max)  # clip and copy
+    # Convert 180 degree range into 360 to match periodicity of hue.
+    ori_ = orientation * 360 / np.pi
+    overlay_final = np.zeros_like(retardance)
+
+    # FIX ME: this binning code leads to artifacts.
```

Comment:
not sure if this is still relevant. otherwise remove these lines

Reply:
Good call. Removed.